.github/workflows/e2e-tests.yaml

```
@@ -21,18 +21,28 @@ env:
 jobs:
   test-e2e:
     runs-on: ubuntu-latest
-
+    strategy:
+      matrix:
+        kubectl_version:
+          - v1.27.4
+          - v1.28.0
     steps:
     # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
     - name: Checkout repo
       uses: actions/checkout@v2
 
+    - name: kubectl install
+      uses: azure/setup-kubectl@v3
+      with:
+        version: ${{ matrix.kubectl_version }}
+
     # Creates KinD with using k8s versions from the matrix above
     - name: Set up kind with K8s version v1.22.4
       uses: engineerd/setup-kind@v0.5.0
       with:
         version: "v0.11.1"
         image: kindest/node:v1.22.4
```

We probably should also adjust the node image version to the kubectl version, in which case we should rename the matrix variable name to `kubernetes version` or just `kubernetes`.

```suggestion
        image: kindest/node:${{ matrix.kubectl_version }}
```

---

This gets trickier, since KinD doesn't distribute all versions (`v1.27.4` is not available for example), and also not all KinD images are compatible with newer versions. I'll give it a shot, but we may need to manage the versions separately.